tqdm/tests/tests_completion.py

```diff
@@ -0,0 +1,34 @@
+import re
+import tqdm
+import tqdm.cli
+
+RE_OPT = re.compile(r'(\w+)  :', flags=re.M)
+
+
+def doc2opt(doc):
+    return ('--' + i for i in RE_OPT.findall(doc))
+
+
+def test_autogenerate_cli():
+    # CLI options
+    options = {'-h', '--help', '-v', '--version'}
+    for doc in (tqdm.tqdm.__init__.__doc__, tqdm.cli.CLI_EXTRA_DOC):
+        options.update(doc2opt(doc))
+    options.difference_update(
+        '--' + i for i in ('name',) + tqdm.cli.UNSUPPORTED_OPTS)
+
+    # All options
+    # Add new CLI options to this list
+    all_opts = {
+            '-h', '-v', '--help', '--version',
+            '--desc', '--total', '--leave', '--ncols',
+            '--mininterval', '--maxinterval', '--miniters',
+            '--ascii', '--disable', '--unit', '--unit_scale',
+            '--dynamic_ncols', '--smoothing', '--bar_format',
+            '--initial', '--position', '--postfix', '--unit_divisor',
+            '--bytes', '--write_bytes', '--lock_args', '--nrows',
+            '--delim', '--buf_size', '--manpath', '--log',
+            '--comppath'
+        }
+
+    assert options == all_opts
```

Comment:
this file really isn't required - it's just a re-implementation rather than a test.

Reply:
Yeah, it basically is. Do you recommend any other way to add a test for this?

Reply:
If you take a look at the upstream branch I've added https://github.com/tqdm/tqdm/commit/57ecc68afa82aea8d147661cd12a709696c3efef. I suppose you could check that the output includes a small stable subset of options. The only super essential ones I think are (`--help`, `--desc`, `--total`, `--leave`, `--ncols`, `--ascii`, `--dynamic_ncols`, `--position`, `--bytes`, `--nrows`, `--delim`, `--manpath`, `--comppath`).